gcloud-java-bigquery/src/test/java/com/google/gcloud/bigquery/BigQueryImplTest.java

```diff
@@ -0,0 +1,982 @@
+
+  // Dataset options
+  private static final BigQuery.DatasetOption DATASET_OPTION_FIELDS =
+      BigQuery.DatasetOption.fields(BigQuery.DatasetField.ACCESS, BigQuery.DatasetField.ETAG);
+
+  // Dataset list options
+  private static final BigQuery.DatasetListOption DATASET_LIST_ALL =
+      BigQuery.DatasetListOption.all();
+  private static final BigQuery.DatasetListOption DATASET_LIST_PAGE_TOKEN =
+      BigQuery.DatasetListOption.startPageToken("cursor");
+  private static final Map<BigQueryRpc.Option, ?> DATASET_LIST_OPTIONS = ImmutableMap.of(
+      BigQueryRpc.Option.ALL_DATASETS, true,
+      BigQueryRpc.Option.PAGE_TOKEN, "cursor");
+
+  // Dataset delete options
+  private static final BigQuery.DatasetDeleteOption DATASET_DELETE_CONTENTS =
+      BigQuery.DatasetDeleteOption.deleteContents();
+  private static final Map<BigQueryRpc.Option, ?> DATASET_DELETE_OPTIONS = ImmutableMap.of(
+      BigQueryRpc.Option.DELETE_CONTENTS, true);
+
+  // Table options
+  private static final BigQuery.TableOption TABLE_OPTION_FIELDS =
+      BigQuery.TableOption.fields(BigQuery.TableField.SCHEMA, BigQuery.TableField.ETAG);
+
+  // Table list options
+  private static final BigQuery.TableListOption TABLE_LIST_MAX_RESULTS =
+      BigQuery.TableListOption.maxResults(42L);
+  private static final BigQuery.TableListOption TABLE_LIST_PAGE_TOKEN =
+      BigQuery.TableListOption.startPageToken("cursor");
+  private static final Map<BigQueryRpc.Option, ?> TABLE_LIST_OPTIONS = ImmutableMap.of(
+      BigQueryRpc.Option.MAX_RESULTS, 42L,
+      BigQueryRpc.Option.PAGE_TOKEN, "cursor");
+
+  // TableData list options
+  private static final BigQuery.TableDataListOption TABLE_DATA_LIST_MAX_RESULTS =
+      BigQuery.TableDataListOption.maxResults(42L);
+  private static final BigQuery.TableDataListOption TABLE_DATA_LIST_PAGE_TOKEN =
+      BigQuery.TableDataListOption.startPageToken("cursor");
+  private static final Map<BigQueryRpc.Option, ?> TABLE_DATA_LIST_OPTIONS = ImmutableMap.of(
+      BigQueryRpc.Option.MAX_RESULTS, 42L,
+      BigQueryRpc.Option.PAGE_TOKEN, "cursor");
+
+  // Job options
+  private static final BigQuery.JobOption JOB_OPTION_FIELDS =
+      BigQuery.JobOption.fields(BigQuery.JobField.USER_EMAIL);
+
+  // Job list options
+  private static final BigQuery.JobListOption JOB_LIST_OPTION_FIELD =
+      BigQuery.JobListOption.fields(BigQuery.JobField.STATISTICS);
+  private static final BigQuery.JobListOption JOB_LIST_ALL_USERS =
+      BigQuery.JobListOption.allUsers();
+  private static final BigQuery.JobListOption JOB_LIST_STATE_FILTER =
+      BigQuery.JobListOption.stateFilter(JobStatus.State.DONE, JobStatus.State.PENDING);
+  private static final Map<BigQueryRpc.Option, ?> JOB_LIST_OPTIONS = ImmutableMap.of(
+      BigQueryRpc.Option.ALL_USERS, true,
+      BigQueryRpc.Option.STATE_FILTER, ImmutableList.of("done", "pending"));
+
+  // Query Results options
```

Comment:
If you think it's useful, we could add the max results and start page token options here